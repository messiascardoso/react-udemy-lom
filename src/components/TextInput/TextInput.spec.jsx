import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TextInput } from ".";

describe("<TextInput />", () => {
  it("should have a value of searchValue", () => {
    const fn = jest.fn();
    const { debug } = render(
      <TextInput handleChange={fn} searchValue={"testando"} />
    );
   // debug();
    const input = screen.getByLabelText("Pesquisar");
    expect(input).toHaveAttribute("type", "search");
    expect(input.value).toBe("testando");
  });

  it("should call handleChange function on each keu pressed", () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} />);
    const input = screen.getByLabelText("Pesquisar");
    const value = "valor digitado no input";
    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should matsh snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange=
      {fn}/>);
    expect(container).toMatchSnapshot();  
  });
});
