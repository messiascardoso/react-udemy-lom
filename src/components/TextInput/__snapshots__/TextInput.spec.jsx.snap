// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should matsh snapshot 1`] = `
<div>
  <div
    class="textInput"
  >
    <label
      class=""
      for="textInput"
    >
      Pesquisar
    </label>
    <input
      id="textInput"
      type="search"
      value="testando"
    />
  </div>
</div>
`;
